contracts/Hub.sol

```diff
@@ -0,0 +1,138 @@
+     * @param amount amount of token that gets minted
+     * @param tickerSymbol identifier of which token gets minted
+     */
+    function mintToken(address receiver, uint256 amount, string memory tickerSymbol) public onlyAuthorizedAddresses{
+        //TODO: mintToken for a specific synthAsset. 
+        //How do we interact with contracts we don't have the interface for
+        //-> every synthAsset contract
+        //maybe heritage??
+        minterContract.mint(receiver, amount); 
+    }
+
+    /**
+     * @notice sets the Minter contract address
+     * @param _minterAddress address of Minter contract
+     */
+    function setMinterContract(address _minterAddress) public onlyOwner{
+        minterContract = Minter(_minterAddress);
+    }
+
+    /**
+     * @notice sets the Fund contract address
+     * @param _fundAddress address of Fund contract
+     */
+    function setFundContract(address _fundAddress) public onlyOwner{
+        fundContract = Fund(_fundAddress);
+    }
+
+    /**
+     * @notice sets the Storage contract address
+     * @param _storageAddress address of Storage contract
+     */
+    function setStorageContract(address _storageAddress) public onlyOwner{
+        storageContract = Storage(_storageAddress);
+    }
+
+    /**
+     * @notice checks the collateral ratio of an address.
+     * @dev for all assets combined. No individual / isolated positions for now.
+     * @dev price checking of colllateral too -> volatile collateral / depegging of stable collateral
+     * @param addr address of user whom collateral ratio is to be checked.
+     * @param collateralTickerSymbol identifier of token used for collateral#
+     * @return collateral ratio
+     */
+    function getCollateralRatio(address addr, string memory collateralTickerSymbol) public view returns (uint256) {
+         
+         //Check amount funded
+         uint256 amountFunded = fundContract.getAmountFundedByAddress(msg.sender);
+         uint256 collateralPrice = storageContract.getAssetPrice(collateralTickerSymbol);
+         uint256 collateralValue = amountFunded * collateralPrice;
+
+         //Check assets minted
+         string[] memory assetsMinted = storageContract.getAssetListOfUser(addr);
+         // Total value of minted assets
+         uint256 totalValueMinted = 0;
+         //Sum up total value of minted assets
+         for(uint i = 0; i < assetsMinted.length; i++) {
+             uint256 assetAmount = storageContract.getAssetAmountOfUser(addr, assetsMinted[i]);
+             uint256 assetPrice = storageContract.getAssetPrice(assetsMinted[i]);
+             uint256 assetValue = assetAmount * assetPrice;
+             totalValueMinted += assetValue;
+         }
+         //return collateral / totalValueMinted 
+         return collateralValue  / totalValueMinted;
+    }
+
+}
+
+interface Minter {
+    function mint(address to, uint256 amount) external;
+}
+
+interface Fund {
```

Comment:
quoting other comment "let's call the contract e.g. CollateralFunds , so that the name of the contract is clear in itself, that it handles deals with saving collateral."